shared-libs/tombstone-utils/src/tombstone-utils.js

```diff
@@ -1,141 +1,11 @@
-        return doc;
-      }
-
-      // we've received a delete stub doc
-      const previousRevision = getPreviousRev(doc._revisions);
-      if (!previousRevision) {
-        return doc;
-      }
-
-      return DB.get(change.id, { rev: previousRevision });
-    });
-};
-
-// CouchDB/Fauxton deletes don't include doc fields in the deleted revision
-const isDeleteStub = function(doc) {
-  // determines if array2 is included in array1
-  const arrayIncludes = function(array1, array2) {
-    return array2.every(function(elem) {
-      return array1.indexOf(elem) !== -1;
-    });
-  };
-
-  return arrayIncludes(COUCHDB_TOMBSTONE_PROPERTIES, Object.keys(doc)) && !!doc._deleted;
-};
-
-// Returns previous rev
-// @param {Object} revisions - doc _revisions
-const getPreviousRev = function(revisions) {
-  if (revisions && revisions.start > 1 && revisions.ids && revisions.ids.length > 1) {
-    return [revisions.start - 1, '-', revisions.ids[1]].join('');
-  }
-  return false;
-};
+const generateTombstoneId = (id, rev) => [id, rev, TOMBSTONE_TYPE].join(TOMBSTONE_ID_SEPARATOR);
 
 module.exports = {
   regex: new RegExp('(.*)' + TOMBSTONE_ID_SEPARATOR + '(.*)' + TOMBSTONE_ID_SEPARATOR + TOMBSTONE_TYPE),
-
-  extractStub: function(tombstoneId) {
-    const match = tombstoneId.match(module.exports.regex);
-    return match && { id: match[1], rev: match[2] };
-  },
-
-  extractDoc: function(tombstoneDoc) {
-    return tombstoneDoc && tombstoneDoc.tombstone;
-  },
-
-  isTombstoneId: function(tombstoneId) {
-    return module.exports.regex.test(tombstoneId);
-  },
-
-  processChange: function (Promise, DB, change, logger) {
-    if (!logger) {
-      logger = console;
-    }
-    if (!change.deleted) {
-      throw new Error('Tombstone: received non-deletion change to tombstone');
-    }
-
-    return getDoc(Promise, DB, change)
-      .then(function(doc) {
-        return doc.type !== TOMBSTONE_TYPE && saveTombstone(DB, doc, change, logger);
-      })
-      .catch(function(err) {
-        logger.error(`Tombstone: could not process doc id: ${change.id} seq: ${change.seq}: %o`, err);
-        throw err;
-      });
-  },
-
-  // generates a copy of the deletion change of the original document
-  generateChangeFromTombstone: function(tombstoneChange, includeDoc) {
-    const stub = module.exports.extractStub(tombstoneChange.id);
-    const change = {
-      id: stub.id,
-      seq: tombstoneChange.seq,
-      changes: [{ rev: stub.rev }],
-      deleted: true
-    };
-
-    if (includeDoc && tombstoneChange.doc && tombstoneChange.doc.tombstone) {
-      change.doc = tombstoneChange.doc.tombstone;
-    }
-
-    return change;
-  },
-
+  isTombstoneId: tombstoneId => module.exports.regex.test(tombstoneId),
   generateTombstoneId: generateTombstoneId,
-
   getTombstonePrefix: id => `${id}${TOMBSTONE_ID_SEPARATOR}`,
-
 };
-
-// exposed for testing
-module.exports._isDeleteStub = isDeleteStub;
-module.exports._getPreviousRev = getPreviousRev;
```

Comment:
Is it even worth having the shared-lib any more?

Reply:
I thought about removing it entirely. Let me see how much it's actually used once I remove code in _all_docs and db-doc.

Reply:
It turns out we ended up only using one function from the lib, that determined whether a doc is a tombstone depending on the format of the doc _id. 
This is still used by the API and Sentinel changes watchers to skip processing tombstones. Tombstones will continue to exist in old deployments and there are chances they get edited (or deleted) at some point. I'm partially tempted to keep this one function _somewhere_. What do you think?

Reply:
Let's leave it for now. I can't think of anywhere else to put it...